ocaml/xapi/xapi_host.ml

```diff
@@ -2246,20 +2246,36 @@ let migrate_receive ~__context ~host ~network ~options =
            ( Api_errors.host_cannot_attach_network
            , [Ref.string_of host; Ref.string_of network] ))
   in
-  let ip = Db.PIF.get_IP ~__context ~self:pif in
-  ( if String.length ip = 0 then
-      match Db.PIF.get_ip_configuration_mode ~__context ~self:pif with
-      | `None ->
-          raise
-            (Api_errors.Server_error
-               (Api_errors.pif_has_no_network_configuration, [Ref.string_of pif]))
-      | `DHCP ->
-          raise
-            (Api_errors.Server_error
-               (Api_errors.interface_has_no_ip, [Ref.string_of pif]))
-      | _ ->
-          failwith "No IP address on PIF"
+  let primary_address_type =
+    Db.PIF.get_primary_address_type ~__context ~self:pif
+  in
+  let ip, configuration_mode =
+    match primary_address_type with
+    | `IPv4 ->
+      ( Db.PIF.get_IP ~__context ~self:pif,
+      Db.PIF.get_ip_configuration_mode ~__context ~self:pif )
+    | `IPv6 ->
+      ( List.hd (Db.PIF.get_IPv6 ~__context ~self:pif),
+      Db.PIF.get_ipv6_configuration_mode ~__context ~self:pif )
+  in
+  ( if ip = "" then
+    match configuration_mode with
+    | `None ->
+      raise
+        (Api_errors.Server_error
+          (Api_errors.pif_has_no_network_configuration, [Ref.string_of pif]))
+    | `DHCP ->
+      raise
+        (Api_errors.Server_error
+          (Api_errors.interface_has_no_ip, [Ref.string_of pif]))
+    | _ ->
+      failwith "No IP address on PIF"
```

Comment:
This is a weird case actually. It's only for mode=static without IP, which is "impossible" by construction, and therefore and internal error. However, we might as well raise the error above here as well... Then you can collapse this and the previous case under a single `| _ ->`.